### DISCUSSION TITLE
Access denied for user

### DISCUSSION THREAD

**Discussion by marceltc** [firefly-iii/firefly-iii#5412] [Firefly III] [answered]
So I decided to move my database from a local MySQL 8.0 database to a remote MariaDB 10.5 database, but Firefly III is not able to access the remote db. I was able to successfully copy over the db using `mysqldump` and it looks ok. I created a database user and password and tested from a MySQL client on the Firefly server that I was able to login successfully to the remote db.

However, Firefly III still gives a 500 error page saying `Could not poll the database: SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'(my_MariaDB_server_hostname)' (using password: YES) (SQL: select 'id', 'name', 'data' from 'configuration' where 'name' = is_demo_site and 'configuration'.'deleted_at' is null limit 1) `

I have ensured that:

1. The database information and credentials are correct in the .env file (DB_HOST is the correct IP, DB_PORT is correct `3306`, username and password both correct, DB_CONNECTION is `mysql`)
2. I can log into the remote Firefly database from a client on the Firefly III server using the above credentials and information
3. I have tried running `php artisan migrate --seed`, which gives the error below `SQLSTATE[HY000] [1045] Access denied for user 'firefly'@'(my_MariaDB_server_hostname)' (using password: YES) (SQL: select * from information_schema.tables where table_schema =  
   firefly and table_name = migrations and table_type = 'BASE TABLE')`
4. I've restarted the entire server to make sure, but still no dice

**Comment by JC5** [comment, marked as answer]
1. Does the password contain special characters?
2. whats the name of the env variable you use for the password?

**Reply by marceltc** [reply]
Turns out there was a `#` in my password 😶‍🌫️ . Removing that character fixed it! Thank you @JC5 !

**Reply by JC5** [reply]
Ahhhh yes that one can be tricky! In `.env` files it's a comment 😬 .

Glad to hear it's fixed!
